From 0a309f63b9116afe269d08b269e18e65d0ff9c83 Mon Sep 17 00:00:00 2001
From: Dan Nica <dan.nica@nxp.com>
Date: Mon, 1 Jun 2020 23:43:53 +0300
Subject: [PATCH 143/269] s32g: Ensure BL2 does not overwrite BootROM data in
 SRAM

On uSDHC boot, BootROM uses 4KB starting from 0x343ff000 for ADMA
descriptors.

Issue: ALB-4935
Upstream-Status: Pending 

Signed-off-by: Dan Nica <dan.nica@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/s32g/include/platform_def.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/plat/s32g/include/platform_def.h b/plat/s32g/include/platform_def.h
index 20019aeae..df064ac3c 100644
--- a/plat/s32g/include/platform_def.h
+++ b/plat/s32g/include/platform_def.h
@@ -135,9 +135,8 @@
  * enough to prevent overflowing onto the adjacent SRAM image. Handle with care,
  * wear a helmet and compile with -Os.
  */
-
-/* this may be a bit too relaxed */
-#define BL2_LIMIT		(S32G_SRAM_END - 1)
+#define BOOTROM_ADMA_RSRVD_BASE	(0x343ff000)
+#define BL2_LIMIT		(BOOTROM_ADMA_RSRVD_BASE - 1)
 
 #define BL31SRAM_BASE		BL2_BASE
 #define BL31SRAM_MAX_PAGES	50
-- 
2.17.1

